integration/crd.go

```diff
@@ -16,38 +16,90 @@ limitations under the License.
 package integration
 
 import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+	"io/ioutil"
+	"os"
 	"os/exec"
 	"testing"
 	"time"
 
 	integration_util "github.com/grafeas/kritis/pkg/kritis/integration_util"
+	"github.com/grafeas/kritis/pkg/kritis/testutil"
 	"k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/util/wait"
 )
 
+const (
+	testAttesationAuthority = `apiVersion: kritis.grafeas.io/v1beta1
+kind: AttestationAuthority
+metadata:
+  name: test-attestor
+spec:
+  noteReference: v1alpha1/projects/krits-int-test
+  privateKeySecretName: %s
+  publicKeyData: %s`
+)
+
+// Secret name for test-attestor
+var aaSecret = "test-attestor"
+
 // CRDs is a map of CRD type to names of the expected CRDs to create.
 var crdNames = map[string]string{
 	"imagesecuritypolicies.kritis.grafeas.io": "my-isp",
 }
 
 var crdExamples = []string{
-	// TODO(aaron-prindle) add back attestation-authority-example.yaml
-	// "attestation-authority-example.yaml",
 	"image-security-policy-example.yaml",
 }
 
+func createAttestationAuthority(t *testing.T, ns string) {
+	t.Helper()
+	// Generate a key value pair
+	pubKey, privKey := testutil.CreateKeyPair(t, aaSecret)
+	// get the base encoded value for public key.
+	pubKeyEnc := base64.StdEncoding.EncodeToString([]byte(pubKey))
+
+	// create a tmp dir for keys
+	d, err := ioutil.TempDir("", "_keys")
+	if err != nil {
+		t.Fatalf("unexpected error %s", err)
+	}
+	defer os.RemoveAll(d)
+
+	// create file with contents in dir
+	pubFile := createFileWithContents(t, d, pubKey)
+
+	privFile := createFileWithContents(t, d, privKey)
+
+	// Finally create a kubernetes secret.
+	cmd := exec.Command("kubectl", "create", "secret", "generic", aaSecret,
+		fmt.Sprintf("--from-file=public=%s", pubFile),
+		fmt.Sprintf("--from-file=private=%s", privFile),
+		"-n", ns)
+	if _, err := integration_util.RunCmdOut(cmd); err != nil {
+		t.Fatalf("unexpected error %s", err)
+	}
+
+	// Create the Attestation authority
+	createAA(t, ns, pubKeyEnc)
```

Comment:
I'm not going to ask you to fix this all now, but we should never be at the point where we are nesting test helpers. I'm less strict about integration tests versus unit tests, but I believe that we are now heading down a path that obscures the connection between the test failure and the conditions which led to it. Some good rules of thumb:

- Test helper functions should never call other test helper functions. Doing so makes it difficult to follow the code path. Test readability is more critical than code readability.

- Test helpers should not interact with the code under test. This helper does not, but getAttestations() seems to fall into this category.

- Never pass testing.T into functions when the standard Go pattern of returning an error is sufficient. 

More context: http://go/gocomments#TOC-Assert

Reply:
ahh. thanks! we do that many places everywhere to reduce duplication and work around checking error and failing.